kmod/uoa/uoa.c

```diff
@@ -939,6 +939,46 @@ static struct nf_hook_ops uoa_nf_hook_ops6[] __read_mostly = {
     },
 };
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,3,0)
+static int __net_init __uoa_init(struct net *net)
+{
+    int ret = -ENOMEM;
+
+    ret = nf_register_net_hooks(net, uoa_nf_hook_ops, ARRAY_SIZE(uoa_nf_hook_ops));
+    if (ret < 0)
+        goto ops_hook_fail;
+
+    ret = nf_register_net_hooks(&init_net, uoa_nf_hook_ops6,
```

Comment:
why init_net for ipv6?

Reply:
Ahhh. Sorry for late.
I just added the supporting for ipv6, but only tested the patch under ipv4.
So, if you don't like it. I can remove the supporting for ipv6.

Reply:
> Ahhh. Sorry for late. I just added the supporting for ipv6, but only tested the patch under ipv4. So, if you don't like it. I can remove the supporting for ipv6.

i mean it should be `ret = nf_register_net_hooks(net, uoa_nf_hook_ops6,`